internal/controller/prefectserver_controller_test.go

```diff
@@ -1548,6 +1566,18 @@ var _ = Describe("PrefectServer controller", func() {
 					NamespacedName: name,
 				})
 				Expect(err).NotTo(HaveOccurred())
+
+				// Set the second migration Job to be complete
+				_, _, desiredMigrationJob = controllerReconciler.prefectServerDeployment(prefectserver)
+				migrateJob = &batchv1.Job{}
+				Eventually(func() error {
+					return k8sClient.Get(ctx, types.NamespacedName{
+						Namespace: namespaceName,
+						Name:      desiredMigrationJob.Name,
+					}, migrateJob)
+				}).Should(Succeed())
+				migrateJob.Status.Succeeded = 1
+				Expect(k8sClient.Status().Update(ctx, migrateJob)).To(Succeed())
```

Comment:
kind of annoying, but looking up a `Job` by name means i have to re-run the manifest generator every time to create the hashed name, which is based on the manifest

Reply:
Yeah, I remember thinking about using `.metadata.generation` from the CR since that should be unique and change each time the CR changes. But not sure if there are any drawbacks to that approach.